tools/autograd/templates/Functions.cpp

```diff
@@ -274,6 +274,42 @@ Tensor prod_safe_zeros_backward(const Tensor &grad, const Tensor& inp, int64_t d
   return grad * (exclusive_normal * exclusive_reverse);
 }
 
+Tensor nanprod_backward(const Tensor& grad, const Tensor& input, const Tensor& result) {
+  if (input.dim() == 0) {
+    return grad * input.isnan().logical_not();
+  }
+  Tensor zero_idx = (input == 0).nonzero();
+  if (zero_idx.numel() == 0) {
+    Tensor nans_replace = at::where(at::isnan(input), at::ones_like(input), input);
```

Comment:
Aside: it would be nice to have nan_to_num for this. Maybe a comment would be helpful?

Reply:
Is there a more efficient way to do this? In particular, can you not just use a tensor of a single element instead of producing a tensor of the same size as input with `at::ones_like?` I would expect the single element tensor to broadcast.